pkg/ctl/cmdutils/gitops.go

```diff
@@ -249,16 +250,114 @@ func (l *GitOpsConfigLoader) Load() error {
 		return l.validateWithoutConfigFile()
 	}
 
+	// The reference to ClusterConfig should only be reassigned if ClusterConfigFile is specified
+	// because other parts of the code store the pointer locally and access it directly instead of via
+	// the Cmd reference
 	var err error
+	if l.cmd.ClusterConfig, err = eks.LoadConfigFromFile(l.cmd.ClusterConfigFile); err != nil {
+		return err
+	}
+
+	meta := l.cmd.ClusterConfig.Metadata
+	if meta == nil {
+		return ErrMustBeSet("metadata")
+	}
+
+	for f := range l.flagsIncompatibleWithConfigFile {
```

Comment:
granted, `cmdutils` is pretty sketchy, but can this be handled in one place in the code without too much refactoring?

Reply:
> `cmdutils` is pretty sketchy

biggest understatement ever 😂 

I think I intended to do something with several bits in that file, but then it fell off my radar (read: I got tired). Can definitely take another look at this bit tho.